code/modules/admin/tabs/admin_tab.dm

```diff
@@ -421,6 +421,41 @@
 	remove_verb(src, admin_verbs_hideable)
 	add_verb(src, /client/proc/enable_admin_verbs)
 
+/client/proc/strip_all_in_view()
+	set name = "Strip All"
+	set category = "Admin.InView"
+	set hidden = TRUE
+
+	if(!admin_holder || !(admin_holder.rights & R_MOD))
+		to_chat(src, "Only administrators may use this command.")
+		return
+
+	if(tgui_alert(src, "This will strip ALL mobs within your view range. Are you sure?", "Confirmation", list("Yes", "Cancel")))
+		return
+
+	var/strip_self = tgui_alert(src, "Do you want to strip yourself as well?", "Confirmation", list("Yes", "No") == "Yes")
+
+	// These for loops are basically exactly the same, however the "if strip self" was done outside of loop for (minor) performance reasons
+	if(strip_self)
+		for(var/mob/living/current_mob in view())
+			for (var/obj/item/current_item in current_mob)
+				//no more deletion of ID cards
+				if(istype(current_item, /obj/item/card/id))
+					continue
+				qdel(current_item)
+
+	else
+		for(var/mob/living/current_mob in view())
+			if(usr == current_mob)
```

Comment:
You don't need two separate loops to accommodate strip_self. Just have one loop that has a check `if(strip_self && usr == current_mob)